runtime/karura/src/lib.rs

```diff
@@ -1804,10 +1837,29 @@ impl Convert<MultiLocation, Option<CurrencyId>> for CurrencyIdConvert {
 				parents: 1,
 				interior: X1(Parachain(parachains::phala::ID)),
 			} => Some(Token(PHA)),
+			// adapt for reanchor canonical location: https://github.com/paritytech/polkadot/pull/4470
+			MultiLocation {
+				parents: 0,
+				interior: X1(GeneralKey(key)),
+			} => match &key[..] {
+				#[cfg(feature = "integration-tests")]
+				parachains::bifrost::BNC_KEY => Some(Token(BNC)),
+				key => {
+					if let Ok(currency_id) = CurrencyId::decode(&mut &*key) {
+						match currency_id {
+							Token(KAR) | Token(KUSD) | Token(LKSM) => Some(currency_id),
```

Comment:
do we need `TAI`

Reply:
good catch, I think `TAI` can be also consider as our parachain side token, then we should add it.

Reply:
I checked mandala and acala runtime, `TAI` is only used in `ExistentialDeposits`, but not in `CurrencyIdConvert`, is it by design or a mistake? @xlc